### PR TITLE
Refactor `df.info()` for better readability

### PR DESCRIPTION
**Pull request by V1NAY8**
This PR optimizes `df.info` where `self.columns` and its `len()` is called like so many times, which always calculates and fetches it back. 

So, call it once, re use it

Also converted `_get_columns` directly to `columns` method with property.

@sethmlarson  take a look :)

### CODE REVIEW

File: eland/dataframe.py [hunk 1]
```diff
@@ -844,32 +848,31 @@ def info(
 
         lines = [str(type(self)), self._index_summary()]
 
-        if len(self.columns) == 0:
+        cols: pd.Index = self.columns
```
Comment (sethmlarson) [inline eland/dataframe.py:851]: nit: Can we name these variables `columns` and `number_of_columns` for better readability?

File: eland/dataframe.py [hunk 2]
```diff
@@ -808,7 +807,12 @@ def _index_summary(self):
         return f"{name}: {len(self)} entries{index_summary}"
 
     def info(
-        self, verbose=None, buf=None, max_cols=None, memory_usage=None, null_counts=None
+        self,
+        verbose: Optional[bool] = None,
+        buf: Optional[StringIO] = None,
+        max_cols: Optional[int] = None,
+        memory_usage: Optional[bool] = None,
+        show_counts: Optional[bool] = None,
```
Comment (sethmlarson) [inline eland/dataframe.py:815]: The change from `null_counts` to `show_counts` isn't explained anywhere?

File: eland/dataframe.py [hunk 3]
```diff
@@ -844,32 +848,31 @@ def info(
...
+
+        exceeds_info_cols = col_count > max_cols
 
         # From pandas.DataFrame
         def _put_str(s, space) -> str:
             return f"{s}"[:space].ljust(space)
 
         def _verbose_repr() -> None:
-            lines.append(f"Data columns (total {len(self.columns)} columns):")
+            lines.append(f"Data columns (total {len(cols)} columns):")
```
Comment (sethmlarson) [inline eland/dataframe.py:875]: Can we use `number_of_columns` here? Also add `nonlocal number_of_columns` at the top of the function.

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Thanks for this, probably doesn't improve performance a ton since it's not making a call to Elasticsearch? I appreciate the `@property` fix though, if you find those elsewhere please fix them too.

Few comments for you:

**Comment by sethmlarson** [comment]
jenkins test this please

**Comment by V1NAY8** [comment]
So, `self.columns` will always go to the capability matrix which is pandas dataframe which contains all the information of field mappings. 

I thought why query the dataframe,  every time when we can query it once and use the result again.

File: eland/dataframe.py (hunk 2 above)
Comment (V1NAY8) [reply eland/dataframe.py:815]: To keep inline with pandas => https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.info.html

They have deprecated `null_counts` and replaced it with `show_counts`

**Comment by sethmlarson** [comment]
jenkins test this please

**Review by sethmlarson — APPROVED** [review]
LGTM

**Pull request merged**